src/pages/index.js

```diff
@@ -51,37 +63,136 @@ const profileEditModal = new PopupWithForm(
   handleProfileFormSubmit
 );
 
+const editAvatarModal = new PopupWithForm(
+  "#change-profile-picture-modal",
+  handleAvatarFormSubmit //create a function of this name that runs after I submit the form//
+);
+
+const avatarUpdateValidator = new FormValidator(config, updateAvatarForm);
+
+const deleteConfirmationModal = new PopupWithConfirmation(
+  "#delete-confirmation-modal"
+);
+
 const addCardModal = new PopupWithForm("#add-card-modal", handleCardAddSubmit);
 
 export const previewImageModal = new PopupWithImage("#previewImageModal");
+
 // call setEventListeners
 
 const profileUserInfo = new UserInfo(
   ".profile__title",
-  ".profile__description"
-);
-
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
+  ".profile__description",
+  ".profile__image"
 );
 
 //functions
 
 function handleProfileFormSubmit(formData) {
-  console.log(formData);
-  profileUserInfo.setUserInfo(formData.title, formData.description);
+  //fetch request to change the name and description on the server
+  profileEditModal.setButtonText("Saving...");
+  //profileEditModal.reset();
+  api
+
+    .updateUserInfo(formData)
+    .then((res) => {
+      profileUserInfo.setUserInfo(res.name, res.about);
+      profileEditModal.close();
+      //modal__form.reset();
+    })
+
+    .catch((err) => {
+      console.error(err);
+      alert("Failed to change user info");
+      console.log(formData);
+    })
+    .finally(() => {
+      profileEditModal.setButtonText("Save");
+    });
 }
 
 function handleCardAddSubmit(inputValues) {
   const cardData = { name: inputValues.title, link: inputValues.url };
-  cardSection.addItem(cardData);
-  addFormValidator.resetValidation();
+  addCardModal.setButtonText("Saving...");
+  api
+    .addCard(cardData)
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      addFormValidator.resetValidation();
+    })
+    .finally(() => {
+      addCardModal.setButtonText("Save");
+    });
+  addCardModal.close();
+}
+
+function handleLikeClick(card) {
+  console.log("like card is running");
+  // call the api and pass card.id and card.isLiked
+  api
+    .updateCardLike(card.id, card.isLiked)
+    .then(() => {
+      card.addLike();
+      card.isLiked = true;
+
+      //card.setLikeButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err} Failed to like card`);
+    });
+
+  // after api is successful, we can then update the UI
+}
+
+function handleUnlikeClick(card) {
+  console.log("unlike card is running");
+  api
+    .updateCardLike(card.id, card.isLiked)
+    .then(() => {
+      card.removeLike();
+      card.isLiked = false;
```

Comment:
Needs correction: Please don't change the card properties like this. It can be done only via setters. You should change `isLiked` state inside `removeLike` method